smartsim/settings/batchSettings.py

```diff
@@ -0,0 +1,240 @@
+
+from __future__ import annotations
+import typing as t
+import copy
+
+from smartsim.log import get_logger
+from .._core.utils.helpers import fmt_dict
+from .common import validate_env_vars, validate_args, StringArgument
+from .batchCommand import SchedulerType
+from .translators.batch.pbs import QsubBatchArgTranslator
+from .translators.batch.slurm import SlurmBatchArgTranslator
+from .translators.batch.lsf import BsubBatchArgTranslator
+from .translators import BatchArgTranslator
+from .baseSettings import BaseSettings
+
+logger = get_logger(__name__)
+
+class BatchSettings(BaseSettings):
+    def __init__(
+        self,
+        batch_scheduler: t.Union[SchedulerType, str],
+        scheduler_args: t.Optional[t.Dict[str, t.Union[str,int,float,None]]] = None,
+        env_vars: t.Optional[StringArgument] = None,
+    ) -> None:
+        try:
+            self._batch_scheduler = SchedulerType(batch_scheduler)
+        except KeyError:
+            raise ValueError(f"Invalid scheduler type: {batch_scheduler}")
+        self._arg_translator = self._get_scheduler()
+
+        if env_vars:
+            validate_env_vars(env_vars)
+        self.env_vars = env_vars or {}
+
+        if scheduler_args:
+            validate_args(scheduler_args)
+        self.scheduler_args = scheduler_args or {}
+
+    @property
+    def batch_scheduler(self):
+        return self._batch_scheduler
+
+    @property
+    def arg_translator(self):
+        return self._arg_translator
+
+    @property
+    def scheduler_args(self) -> t.Dict[str, t.Union[int, str, float, None]]:
+        """Retrieve attached batch arguments
+
+        :returns: attached batch arguments
+        """
+        return self._scheduler_args
+
+    @scheduler_args.setter
+    def scheduler_args(self, value: t.Dict[str, t.Union[int, str, float,None]]) -> None:
+        """Attach batch arguments
+
+        :param value: dictionary of batch arguments
+        """
+        self._scheduler_args = copy.deepcopy(value) if value else {}
+
+    def _get_scheduler(self) -> BatchArgTranslator:
+        """ Map the Scheduler to the BatchArgTranslator
+        """
+        if self._batch_scheduler.value == 'slurm':
+            return SlurmBatchArgTranslator()
+        elif self._batch_scheduler.value == 'lsf':
+            return BsubBatchArgTranslator()
+        elif self._batch_scheduler.value == 'pbs':
+            return QsubBatchArgTranslator()
+
+    def scheduler_str(self) -> str:
+        """ Get the string representation of the scheduler
+        """
+        return self.arg_translator.scheduler_str()
+
+    def set_walltime(self, walltime: str) -> None:
+        """Set the walltime of the job
+
+        format = "HH:MM:SS"
+
+        :param walltime: wall time
+        """
+        # TODO check for formatting here
+        args = self.arg_translator.set_walltime(walltime)
+        if args:
+            self.update_scheduler_args(args)
```

Comment:
per offline discussion: it was decide to remove these pass through methods and have the `self.arg_translator` manage its own state so that if can have better context on which/how args to set